test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1292,7 +1295,11 @@ protected final XContentParser createParser(XContentBuilder builder) throws IOEx
      * Create a new {@link XContentParser}.
      */
     protected final XContentParser createParser(XContent xContent, String data) throws IOException {
-        return xContent.createParser(xContentRegistry(), LoggingDeprecationHandler.INSTANCE, data);
+        if (randomBoolean()) {
+            return createParserWithCompatibilityFor(xContent, data, RestApiCompatibleVersion.minimumSupported());
```

Comment:
should this be minimumSupported(), or currentVersion() ? (I think the latter, else we would randomly get current or n-1 parsing ?)

Reply:
actually I thought that we could test it with random. The N-1 parser should also support all the fields declared without the version (should not parse those which are explicitly declared for N only)